projects/interactions/private/HNLFromSpline.cxx

```diff
@@ -137,6 +138,32 @@ void HNLFromSpline::LoadFromMemory(std::vector<char> & differential_data, std::v
     total_cross_section_.read_fits_mem(total_data.data(), total_data.size());
 }
 
+double HNLFromSpline::GetLeptonMass(LI::dataclasses::Particle::ParticleType lepton_type) {
```

Comment:
I think it will work okay. We're using an enum class for ParticleType that is explicitly using the int32_t type under the hood, and I don't think there are any runtime checks that would give us an issue if the underlying value of the in32_t does not match one of the elements of the enum class. I think you can only get compile time errors if you static cast a compile time constant to the enum when the value does not match one of the elements.

Reply:
Oh nvm, I'm dumb. I was just looking at the function signature and not the function body. You're right that it will need to be fixed at some point.